extensions/amp-mplayer/0.1/amp-mplayer.js

```diff
@@ -0,0 +1,480 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Deferred} from '../../../src/utils/promise';
+import {Services} from '../../../src/services';
+import {VideoEvents} from '../../../src/video-interface';
+import {addParamsToUrl} from '../../../src/url';
+import {
+  createFrameFor,
+  isJsonOrObj,
+  objOrParseJson,
+  originMatches,
+  redispatch,
+} from '../../../src/iframe-video';
+import {dev, user, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {
+  fullscreenEnter,
+  fullscreenExit,
+  isFullscreenElement,
+  removeElement,
+} from '../../../src/dom';
+import {getData, listen} from '../../../src/event-helper';
+import {installVideoManagerForDoc} from '../../../src/service/video-manager-impl';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/** @const */
+const TAG = 'amp-mplayer';
+
+/** @implements {../../../src/video-interface.VideoInterface} */
+class AmpMPlayer extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    // Declare instance variables with type annotations.
+    /** @private {Element} */
+    this.iframe_ = null;
+
+    /** @private {string} */
+    this.contentType_ = null;
+
+    /** @private {?string} */
+    this.playerId_ = null;
+
+    /** @private {?string} */
+    this.contentId_ = '';
+
+    /** @private {?string} */
+    this.scannedElement_ = '';
+
+    /** @private {?string} */
+    this.tags_ = '';
+
+    /** @private {?string} */
+    this.minimumDateFactor_ = '';
+
+    /** @private {?string} */
+    this.scannedElementType_ = '';
+
+    /** @private {?Promise} */
+    this.playerReadyPromise_ = null;
+
+    /** @private {?Function} */
+    this.playerReadyResolver_ = null;
+
+    /** @private {?number} */
+    this.readyTimeout_ = null;
+  }
+
+  /**
+   * @param {boolean=} onLayout
+   * @override
+   */
+  preconnectCallback(onLayout) {
+    // Host that serves player configuration and content redirects
+    this.preconnect.url('https://www.oo-syringe.com', onLayout);
+  }
+
+  /** @override */
+  buildCallback() {
+    // Get attributes, assertions of values, assign instance variables.
+    // Build lightweight DOM and append to this.element.
+
+    const {element} = this;
+
+    this.contentType_ = userAssert(
+      element.getAttribute('data-content-type'),
+      'The data-content-type must be specified for <amp-mplayer> %s',
+      element
+    );
+
+    this.contentId_ =
+      this.contentType_ != 'semantic'
+        ? userAssert(
+            element.getAttribute('data-content-id'),
+            'The data-content-id must be specified for %s ' +
+              'data-content-type in <amp-mplayer> %s',
+            this.contentType_,
+            element
+          )
+        : '';
+
+    this.initSemanticFields_();
+
+    const deferred = new Deferred();
```

Comment:
I don't think you need any of the playerReady stuff and timeout. Just do 
`return return this.loadPromise(this.iframe_);`

Reply:
I don't need it for the video-interface implementation?

Reply:
Depends. If you iframe can respond to `postMessages` for `play`, `pause`, etc.. as soon as it is loaded, then no. If your iframe needs to run some initialization code before it can respond to those methods, then yes, you would need to tell us to wait until you are ready and initialized.

Reply:
Our iframe do needs to run some initialization code before it can respond to post messages

Reply:
Sometimes when Im running the example page Im starting to get the post messages before the initiation ended. Can you tell why is this happening or how can we avoid it?

Reply:
Furthermore, what do I need to do in order to make the docking work?
Of course I added all the relevant information and attribute to the example page but it doesnt work. Do I need to make some changes in the code as well? Thanks

Reply:
Hi again. 
I did some changes and now the docking is working (as well as the autoplay).
I have only one problem: when I scroll down the page and the docking is working, I press on the mute button (it works) than again on the unmute button (it works) but from now on it stops responding and sending only unmute commands (although its already unmuted). Can you maybe tell why?

Reply:
You need to trigger `MUTED` and `UNMUTED` events.

Reply:
Already did it and it works now, thanks